plugins/jira-dashboard-backend/src/api.ts

```diff
@@ -109,8 +110,14 @@ export const searchJira = async (
       Accept: 'application/json',
       'Content-Type': 'application/json',
     },
-  }).then(resp => resp.json());
-  return response.issues;
+  });
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch Jira data: ${response.status} ${response.statusText}`,
```

Comment:
If you use the "ResponseError.fromResponse" you will have an object where you can check the statusCode and other fields.

```typescript
   if (!res.ok) {
     throw await ResponseError.fromResponse(res);
   }
```